octavia_f5/controller/worker/sync_manager.py

```diff
@@ -183,6 +183,9 @@ def tenant_update(self, network_id, device=None, selfips=None, loadbalancers=Non
         if not loadbalancers:
             raise exceptions.AS3Exception("No loadbalancers specified for tenant_update")
 
+        # If any of the load balancers is currently being rescheduled, we want to hold back the whole declaration because we don't know whether the load balancer is supposed to exist on this device or not.
+        pass # TODO
+
```

Comment:
Any reason not to use the `provisioning_status`?

Reply:
This seems pretty over-engineered to me, so you want to introduce locks to the status_manager as well as controller_worker? The status is considered a user facing property, nothing to act on.

Reply:
It's not overengineered, it's basically exactly what you proposed, just with all necessary considerations laid out.

Reply:
Unless I misunderstood your initial proposal (the one with `provisioning_status`)